component/src/main/java/org/wso2/extension/siddhi/execution/env/ResourceBatchWindowProcessor.java

```diff
@@ -0,0 +1,367 @@
+                                //schedule the notifier for late event flushing
+                                scheduler.notifyAt(nextEmitTime + LATE_EVENT_FLUSHING_DURATION);
+                            }
+                        }
+                    }
+                }
+                for (Map.Entry<Object, ResourceStreamEventList> entry : groupEventMap.entrySet()) {
+                    int windowLength = ResourceIdentifierStreamProcessor.getResourceCount(resourceName);
+                    List<StreamEvent> streamEventList = null;
+                    if (entry.getValue().isExpired) {
+                        //flushing the late events if the entries already expired and late event flushing interval
+                        //less than the current time.
+                        if ((entry.getValue().expiryTimestamp + LATE_EVENT_FLUSHING_DURATION) < currentTime) {
+                            groupEventMap.remove(entry.getKey());
+                        }
+                    } else {
+                        if (entry.getValue().streamEventList.size() >= windowLength ||
+                                entry.getValue().expiryTimestamp + LATE_EVENT_FLUSHING_DURATION < currentTime) {
+                            //events add into output stream event list and remove entry map entry as
+                            // the LATE_EVENT_FLUSHING_DURATION interval exceed
+                            streamEventList = entry.getValue().streamEventList;
+                            groupEventMap.remove(entry.getKey());
+                        } else if (entry.getValue().expiryTimestamp < currentTime) {
+                            //events add into output stream event list, and wait 'LATE_EVENT_FLUSHING_DURATION' for
+                            // late events before removing the map entry
+                            streamEventList = entry.getValue().streamEventList;
+                            entry.getValue().setExpired(true);
+                        }
+                    }
+                    if (streamEventList != null) {
+                        if (outputExpectsExpiredEvents) {
+                            for (StreamEvent event : entry.getValue().streamEventList) {
+                                StreamEvent toExpireEvent = streamEventCloner.copyStreamEvent(event);
+                                toExpireEvent.setType(StreamEvent.Type.EXPIRED);
+                                eventsToBeExpiredEventChunk.add(toExpireEvent);
+                            }
+                        }
+                        StreamEvent toResetEvent = streamEventCloner.copyStreamEvent(streamEventList.get(0));
+                        toResetEvent.setType(ComplexEvent.Type.RESET);
+                        eventsToBeExpiredEventChunk.add(toResetEvent);
+                        for (StreamEvent event : entry.getValue().streamEventList) {
+                            StreamEvent currentEvent = streamEventCloner.copyStreamEvent(event);
```

Comment:
we dont need to clone here right ? as we are not using this elsewhere

Reply:
yes, we can just use the event from list without cloning, will add that fixes

Reply:
Fixes added : c3a5404a65772aaeeaa276ade0af328f954a0dac